pkg/util/cache/cache.go

```diff
@@ -27,6 +27,9 @@ const (
 type Cache []*cacheShard
 
 func NewCache(maxSize int) Cache {
+	if maxSize < shardsCount {
```

Comment:
You added this because it was making a size-0 shard, I guess? I think the better thing to do is to make the shard count not be fixed at 32, that seems really weird. I'm not sure what this cache is used for, though.